files/external_node_v2.rb

```diff
@@ -196,8 +196,10 @@ def enc(certname)
   end
   res = http.start { |http| http.request(req) }
 
-  raise "Error retrieving node #{certname}: #{res.class}\nCheck Foreman's /var/log/foreman/production.log for more information." unless res.code == "200"
-  res.body
+  return res.body if res.code == '200'
+
+  raise res.class, "Error retrieving node #{certname}: #{res.class}"\
```

Comment:
This line seems to cause the following error if executed:

<pre>
/etc/puppetlabs/puppet/node.rb:201:in `raise': exception class/object expected (TypeError)
        from /etc/puppetlabs/puppet/node.rb:201:in `enc'
        from /etc/puppetlabs/puppet/node.rb:346:in `block in <main>'
        from /usr/share/ruby/timeout.rb:66:in `timeout'
        from /etc/puppetlabs/puppet/node.rb:345:in `<main>'
</pre>

(on Ruby 2.0.0)